src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -862,9 +893,35 @@ private static FunctionResult ItemsResultBuilder(IBinder binder, IFileResolver f
 
         private static TypeSymbol GetFlattenReturnType(TypeSymbol typeToFlatten, IPositionable argumentPosition)
         {
-            static TypeSymbol FlattenUnionOfArrays(UnionType unionType, IPositionable argumentPosition) => UnionOfFlattened(
-                unionType,
-                unionType.Members.Select(typeRef => GetFlattenReturnType(typeRef.Type, argumentPosition)),
+            static TypeSymbol FlattenTuple(TypeSymbol flattenInputType, TupleType tupleType, IPositionable argumentPosition)
```

Comment:
For these "type-math" functions, should we move them into a dedicated helper class? Since they're not specific to the `sys` namespace.